File: DataAccess/Repositories/Modules/Updates/Abstractions/Human.cs
namespace DataAccess.Repositories.Modules.Updates.Abstractions;

public abstract class Human
{
    public Image? Image { get; set; }
    public required Address Address { get; set; }

    public required string FirstName { get; set; }
    public required string MidName { get; set; }
    public required string LastName { get; set; }

    public required DateOnly DateOfBirth { get; set; }

    public required string PhoneNumber { get; set; }

    public required string Email { get; set; }
}

File: website 16/src/classes/human.ts
import { Image, Location } from "./entities";

abstract class Human {
  public ID: string;

  public Avatar: Image;

  public Location: Location;

  public FirstName: string;
  public MidName: string;
  public LastName: string;

  public DateOfBirth: Date;

  public PhoneNumber: string;

  public Email: string;

  public UpdatedAt: Date;
  public CreatedAt: Date;

  constructor(
    id: string,
    avatar: Image,
    location: Location,
    firstName: string,
    midName: string,
    lastName: string,
    dateOfBirth: Date,
    phoneNumber: string,
    email: string,
    updatedAt: Date,
    createdAt: Date,
  ) {
    this.ID = id;

    this.Location = location;

    this.Avatar = avatar;

    this.FirstName = firstName;
    this.MidName = midName;
    this.LastName = lastName;

    this.DateOfBirth = dateOfBirth;

    this.PhoneNumber = phoneNumber;

    this.Email = email;

    this.UpdatedAt = updatedAt;
    this.CreatedAt = createdAt;
  }
}

export { Human };
